demos/unity_states/cs/src/BallStates.cs

```diff
@@ -3,7 +3,7 @@
 using ffi = unity_states.generated.__Internal;
 
 /// NOTE: this class is only ever a shared ref in rust
-public struct BallStates
+public class BallStates
```

Comment:
See point 1 in [this Slack message](https://witag.slack.com/archives/GARELNYKY/p1663292633368609?thread_ts=1663051425.215129&cid=GARELNYKY).

Reply:
Would be good to actually definitively show why this is necessary. Otherwise its indicative of the problem not being understood

Reply:
Yes, it is to work around an [open bug in .NET CLR](https://github.com/dotnet/runtime/issues/6924) where instantiating a struct with a generic (in our case, `Ptr<BallStates>`) causes a `TypeLoadException` at runtime.

Reply:
unfortunate. I guess we can do a class for now.